releasenotes/notes/drop-gateway-alpha.yaml

```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
+area: traffic-management
+releaseNotes:
+- |
+  **Removed** `alpha` Gateway API types by default. They can be explicitly re-enabled with `PILOT_ENABLE_ALPHA_GATEWAY_API=true`.
```

Comment:
More ? What does it mean for a user ( "any GW resource with alpha version will be dropped, user must make sure they don't have any such CR using command ... and convert them by ... "?). 

Is there a good reason for the env variable - user who can't do the migration to beta can delay upgrading until they do this. I'm not sure what will be the behavior of this env is enabled - did we test how it interacts with the beta CRDs ?

Reply:
There is no beta yet. The issue is there is only alpha now, and we are explicitly opting users into alpha. So when their k8s platform changes and they are on old istio, they may have ONLY the Beta TCPRoute CRD. So we will try to read alpha, but its removed, and things break.

Fundamentally, we should never watch Alpha CRDs without explicit opt in. Especially when they are not owned by Istio, as they can and will be removed and break users.